x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/normalization/rule_converters.ts

```diff
@@ -442,7 +446,9 @@ export const convertPatchAPIToInternalSchema = (
       description: nextParams.description ?? existingParams.description,
       ruleId: existingParams.ruleId,
       falsePositives: nextParams.false_positives ?? existingParams.falsePositives,
-      investigationFields: nextParams.investigation_fields ?? existingParams.investigationFields,
+      investigationFields: migrateLegacyInvestigationFields(
+        nextParams.investigation_fields ?? existingParams.investigationFields
+      ),
```

Comment:
Related to https://github.com/elastic/kibana/pull/169061#discussion_r1372377694, I'd consider omitting the migration logic here to make it easier to reason about which routes will migrate data as it comes into the server. If we remove the migration here, I believe the import route will be the only route to modify `investigation_fields` on write. Every other route will simply pass through whatever it receives from the API, which now in 8.11 will always be the new object-based type. PATCH based routes would leave `investigation_fields` un-modified if the field is not specified in the request.
```suggestion
      investigationFields: nextParams.investigation_fields ?? existingParams.investigationFields,
```

Reply:
Updated so now we are only migrating the response and in rule import. PATCH will not migrate unless specified in request.